feature-add-meal/src/main/java/com/kinandcarta/lib/add/meal/viewmodel/AddMealViewModel.kt

```diff
@@ -0,0 +1,100 @@
+package com.kinandcarta.lib.add.meal.viewmodel
+
+import android.location.Location
+import android.location.LocationManager
+import androidx.compose.runtime.setValue
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.kcc.kmmhackathon.shared.MealsSDK
+import com.kcc.kmmhackathon.shared.entity.Meal
+import kotlinx.coroutines.launch
+import java.time.LocalDateTime
+
+data class MealForm constructor(
+    var title: String = "",
+    var additionalInformation: String = "",
+    var quantity: String = "",
+    var isHot: Boolean = true,
+    var availableFrom: LocalDateTime = LocalDateTime.now(),
+    var useBy: LocalDateTime = LocalDateTime.now(),
+    var address: String = "",
+    var location: Location = Location(LocationManager.GPS_PROVIDER)
+)
+
+class AddMealViewModel : ViewModel() {
+    sealed class State() {
+        object Initial: State()
+        object Loading: State()
+        object Success: State()
+        object Failed: State()
+    }
+
+    var state: State by mutableStateOf(State.Initial)
+        private set
+
+    var meal: MealForm by mutableStateOf(MealForm())
+        private set
+
+    private var mealsSDK: MealsSDK
+
+    init {
+        this.mealsSDK = MealsSDK()
+    }
+
+    fun onEditTitle(newValue: String) {
+        meal.title = newValue
+    }
+
+    fun onEditAdditionalInformation(newValue: String) {
+        meal.additionalInformation = newValue
+    }
+
+    fun onEditQuantity(newValue: String) {
+        meal.quantity = newValue
+    }
+
+    fun onEditTemperature(newValue: Boolean) {
+        meal.isHot = newValue
+    }
+
+    fun onEditAvailableFrom(newValue: LocalDateTime) {
+        meal.availableFrom = newValue
+    }
+
+    fun onEditUseBy(newValue: LocalDateTime) {
+        meal.useBy = newValue
+    }
+
+    fun onEditAddress(newValue: String) {
+        meal.address = newValue
+    }
+
+    fun onSubmit() {
+        state = State.Loading
+        viewModelScope.launch {
+            kotlin.runCatching {
+                val meal = Meal(
+                    id = mealsSDK.getUUID().toString(),
+                    name = meal.title,
+                    info = meal.additionalInformation,
+                    quantity = meal.quantity.toInt(),
+                    availableFromDate = meal.availableFrom.toString() ,
```

Comment:
I got an error after submitting a meal with the date time format in list view

`java.lang.NumberFormatException: For input string: 2020-12-06T14:17`

Perhaps we need to use the SharedDate utility that Gareth made to process the date in the post?